data-services/src/test/java/org/pdxfinder/DataImportServiceTests.java

```diff
@@ -0,0 +1,160 @@
+                .thenReturn(new HashSet<>());
+
+        actualSnapshot = dataImportService.getPatientSnapshot(PATIENT, AGE_AT_COLLECTION);
+
+        Assert.assertEquals(PATIENT, actualSnapshot.getPatient());
+        Assert.assertEquals(AGE_AT_COLLECTION, actualSnapshot.getAgeAtCollection());
+    }
+
+    @Test
+    public void Given_GetPatientSnapshot4Param_When_ValidArg_Then_ReturnPSfromPatientWithEqualREF() {
+
+        Patient patientWithSnapshots = new Patient(EXTERNAL_ID, SEX, RACE, ETHNICITY, GROUP);
+
+        patientWithSnapshots.hasSnapshot(SNAPSHOT);
+        SNAPSHOT.setPatient(patientWithSnapshots);
+
+        actualSnapshot = dataImportService.getPatientSnapshot(patientWithSnapshots, AGE_AT_COLLECTION,
+                COLLECTION_DATE, COLLECTION_EVENT, ELAPSED_TIME);
+
+        Assert.assertEquals(SNAPSHOT, actualSnapshot);
+    }
+
+
+    @Test
+    public void Given_GetPatientSnapshot4Param_When_ValidArgAndNoMatchingPS_Then_ReturnNewPSforPatient(){
+
+        actualSnapshot = dataImportService.getPatientSnapshot(PATIENT, AGE_AT_COLLECTION,
+                COLLECTION_DATE, COLLECTION_EVENT, ELAPSED_TIME);
+
+        Assert.assertEquals(PATIENT, actualSnapshot.getPatient());
+        Assert.assertEquals(AGE_AT_COLLECTION, actualSnapshot.getAgeAtCollection());
+        Assert.assertEquals(COLLECTION_DATE, actualSnapshot.getDateAtCollection());
+        Assert.assertEquals(COLLECTION_EVENT, actualSnapshot.getCollectionEvent());
+        Assert.assertEquals(ELAPSED_TIME, actualSnapshot.getDateAtCollection());
+    }
+
+    @Test
+    public void Given_GetPatientSnapshot6Param_When_PatientInDB_Then_ReturnNewPSforPatient(){
+
+        when(patientRepository.findByExternalIdAndGroup(EXTERNAL_ID, GROUP))
+                .thenReturn(PATIENT);
+
+        when(dataImportService.getPatientSnapshot(PATIENT, AGE_AT_COLLECTION))
+                .thenReturn(actualSnapshot);
+
+        actualSnapshot = dataImportService.getPatientSnapshot(EXTERNAL_ID, SEX, RACE,
+                ETHNICITY, AGE_AT_COLLECTION, GROUP);
+
+        Assert.assertEquals(PATIENT, actualSnapshot.getPatient());
+        Assert.assertEquals(AGE_AT_COLLECTION, actualSnapshot.getAgeAtCollection());
+    }
+
+    @Test
+    public void Given_getPatientSnapshot6Param_When_ValidArgAndNoPatientInDB_Then_ReturnMatchingPS(){
+
+        when(patientRepository.findByExternalIdAndGroup(EXTERNAL_ID, GROUP))
+                .thenReturn(null);
+
+         actualSnapshot = dataImportService.getPatientSnapshot(EXTERNAL_ID, SEX, RACE,
+                 ETHNICITY, AGE_AT_COLLECTION, GROUP);
+
+        Assert.assertEquals(EXTERNAL_ID,  actualSnapshot.getPatient().getExternalId());
+        Assert.assertEquals(AGE_AT_COLLECTION,  actualSnapshot.getAgeAtCollection());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void Given_getpatiensnapshot6param_When_externalidisblank_Then_returnnullexception(){
```

Comment:
change test name to match others, same with the below test. Minor  typo here too.
```suggestion
    public void Given_GetPatientSnapshot6Param_When_ExternalIdIsBlank_Then_ReturnNullException(){
```